citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/AuthorityService.java

```diff
@@ -118,6 +182,52 @@ public IAuthorityEntry find(String id) {
         return entries;
     }
 
+
+    public List<IAuthorityEntry> findByName(IUser user, String firstName, String lastName, int page, int pageSize) {
+        Pageable paging = PageRequest.of(page, pageSize);
+        List<IAuthorityEntry> results = authorityRepository.findByUsernameAndNameContainingAndNameContainingOrderByName(
+                user.getUsername(), firstName, lastName, paging);
+        return results;
+    }
+    
+    @Override
+    public Set<IAuthorityEntry> findByNameInDataset(String firstName, String lastName, String citationGroupId,
+            int page, int pageSize, List<String> uris) throws GroupDoesNotExistException {
+        Optional<ICitationGroup> group = groupRepository.findFirstByGroupId(new Long(citationGroupId));
+        if (!group.isPresent()) {
+            throw new GroupDoesNotExistException("Group with id " + citationGroupId + " does not exist.");
+        }
+        Pageable paging = PageRequest.of(page, pageSize);
+        Persons persons = new Persons();
+        List<Persons> personsList = null;
+        if (uris != null && uris.size() > 0) {
+            personsList = personsMongoDao.findPersonsByCitationGroupAndNameLikeAndUriNotIn(
+                    (ICitationGroup) group.get(), firstName, lastName, uris, paging);
+        } else {
+            persons = personsMongoDao.findPersonsByCitationGroupAndNameLike((ICitationGroup) group.get(),
+                    firstName, lastName, paging);
+        }
+        Set<IAuthorityEntry> entries = new HashSet<>();
+        if(persons != null) {
+            personsList.forEach(per -> per.getPersons().forEach(p -> {
```

Comment:
I don't understand what happens here. Why do you check if `persons` is not null and then iterate over `personsList`?